### PR TITLE
Quadrat: Add page templates

### PR DESCRIPTION
**Pull request by scruffian**
#### Changes proposed in this Pull Request:
This moves templates to the block-templates directory. This should mean that the theme works in the Site Editor if you have it enabled, but if you don't then we fall back to the PHP templates. You can test by disabling the Site Editor with this PR: https://github.com/WordPress/gutenberg/pull/31339

#### Related issue(s):
https://github.com/Automattic/themes/issues/3643

### CODE REVIEW

File: quadrat/block-template-parts/404.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <div class="wp-block-group">
 
 <!-- wp:heading {"textAlign":"center","level":1,"fontSize":"large"} -->
-<h1 class="has-text-align-center has-large-font-size">Oops! That page can’t be found.</h1>
+<h1 class="has-text-align-center has-large-font-size">Quadrat Oops! That page can’t be found.</h1>
```
Comment (MaggieCabrera) [inline quadrat/block-template-parts/404.html]: I think this change comes from testing template loading, right? cause I've done the same :D

Comment (scruffian) [reply quadrat/block-template-parts/404.html]: Yeah :)

**Comment by jffng** [comment]
- Need separate page + single templates
- Change index.php to use query block so title of post can appear

File: quadrat/functions.php [hunk 2]
```diff
@@ -19,6 +19,8 @@ function quadrat_support() {
 				'social'  => __( 'Social Links Navigation', 'quadrat' ),
 			)
 		);
+
+		remove_theme_support( 'block-templates' );
```
Comment (jffng) [inline quadrat/functions.php:23]: Why do we want to remove this?

**Comment by jffng** [comment]
I made some revisions to this PR: 

- Index template now uses query block
- Separated singular into page + single templates because they contain different blocks
- Revised single template to include post tags, post date, and post featured image
- Rebased `make/quadrat`
- Removed `index.html` for now while we consider the theme support to disable the site editor 

It's a significant improvement over what we have now, so going to merge and we can iterate.

**Pull request merged**